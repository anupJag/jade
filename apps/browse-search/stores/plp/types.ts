import { Product } from '@jade/ui/organism/types';
import { Filter, PriceFilterRange, ValueFilterRange } from '@jade/ui/molecules/types';

export enum ActionTypes {
  SET_INITIAL_STATE = 'SET_INITIAL_STATE',
}

export type FilterKey =
  | 'price'
  | 'rating'
  | 'discount'
  | 'category'
  | 'shippingTags'
  | 'brand'
  | 'reviewCount';

export type AppliedFilters = {
  price?: PriceFilterRange[];
  rating?: ValueFilterRange[];
};

export type State = {
  products: Product[];
  filters: Filter[];
  appliedFilters: AppliedFilters;
  filteredProducts: Product[];
  productLoaded: boolean;
};

export type SetInitialStateAction = {
  type: ActionTypes.SET_INITIAL_STATE;
  payload: {
    products: Product[];
    filters: Filter[];
  };
};

export type EmptyAction = { type: '' };

export type Action = SetInitialStateAction | EmptyAction;
