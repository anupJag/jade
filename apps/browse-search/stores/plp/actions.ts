import { Action, ActionTypes } from './types';
import { Product } from '@jade/ui/organism/types';
import { Filter } from '@jade/ui/molecules/types';

export const setInitialState = (
  products: Product[],
  filters: Filter[],
  dispatch: React.Dispatch<Action>,
) => {
  dispatch({
    type: ActionTypes.SET_INITIAL_STATE,
    payload: {
      products,
      filters,
    },
  });
};
