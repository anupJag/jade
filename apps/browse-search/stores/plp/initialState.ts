import { State } from './types';

export const initialState: State = {
  products: [],
  filters: [],
  appliedFilters: {},
  filteredProducts: [],
  productLoaded: false,
};
