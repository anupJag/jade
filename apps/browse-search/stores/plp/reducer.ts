import { Action, ActionTypes, State } from './types';
import { initialState } from './initialState';

const plpReducers = (state: State = initialState, action: Action = { type: '' }) => {
  switch (action.type) {
    case ActionTypes.SET_INITIAL_STATE:
      return {
        ...state,
        products: [...action.payload.products],
        filteredProducts: [...action.payload.products],
        filters: [...action.payload.filters],
        appliedFilters: {},
        productLoaded: true,
      };
    default:
      return state;
  }
};

export default plpReducers;
