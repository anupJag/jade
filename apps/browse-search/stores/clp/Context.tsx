/** @context.tsx
 *  This component is a wrapper which will provide
 *  the Global Context to other needy components
 *  */

import React, { FC, useReducer } from 'react';
import { initialState } from './initialState';
import reducer from './reducer';
import { Action, State } from './types';

interface CLPContextProps {
  state: State;
  dispatch: React.Dispatch<Action>;
}

export const CLPContext = React.createContext<CLPContextProps>({
  state: initialState,
} as CLPContextProps);

// eslint-disable-next-line react/prop-types
const CLPContextProvider: FC = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, initialState);
  return <CLPContext.Provider value={{ state, dispatch }}>{children}</CLPContext.Provider>;
};

export default CLPContextProvider;
