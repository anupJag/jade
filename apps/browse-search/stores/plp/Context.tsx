/** @context.tsx
 *  This component is a wrapper which will provide
 *  the Global Context to other needy components
 *  */

import React, { FC, useReducer } from 'react';
import { initialState } from './initialState';
import reducer from './reducer';
import { Action, State } from './types';

interface PLPContextProps {
  state: State;
  dispatch: React.Dispatch<Action>;
}

export const PLPContext = React.createContext<PLPContextProps>({
  state: initialState,
} as PLPContextProps);

// eslint-disable-next-line react/prop-types
const PLPContextProvider: FC = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, initialState);
  return <PLPContext.Provider value={{ state, dispatch }}>{children}</PLPContext.Provider>;
};

export default PLPContextProvider;
