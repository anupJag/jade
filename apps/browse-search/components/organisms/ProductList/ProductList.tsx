import React, { FC, useContext } from 'react';
import { Row } from '@jade/ui/atoms';
import { Alert } from '@jade/ui/molecules';
import { ProductTile } from '@jade/ui/organism';
import { ListItem } from './ProductList.styled';
import { PLPContext } from '../../../stores/plp/Context';

const ProductList: FC = () => {
  const { state } = useContext(PLPContext);
  const { filteredProducts, productLoaded } = state;

  if (!productLoaded) {
    return null;
  }

  const Children = filteredProducts.map(product => (
    <ListItem columns={12} key={product.sku}>
      <ProductTile product={product} />
    </ListItem>
  ));

  if (Children.length > 0) {
    return <Row mb="lg">{Children}</Row>;
  }

  return (
    <Alert
      status="info"
      variant="subtle"
      title="Sorry!"
      description="We could not found any results for given criteria!!!"
    />
  );
};

export default ProductList;
