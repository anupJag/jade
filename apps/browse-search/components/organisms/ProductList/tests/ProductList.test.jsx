import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@jade/ui/themes';
import { PLPContext } from '../../../../stores/plp/Context';

import mockResponse from '../../../../../public/mock/PLP.json';

import ProductList from '../index';

describe('ProductList', () => {
  let ProductListComponent = '';
  const dispatch = jest.fn();

  beforeEach(() => {
    const products = mockResponse.products;
    const mockState = {
      products,
      filteredProducts: products,
    };
    ProductListComponent = render(
      <ThemeProvider>
        <PLPContext.Provider value={{ state: mockState, dispatch }}>
          <ProductList />
        </PLPContext.Provider>
      </ThemeProvider>,
    );
  });

  it('should render products correctly', () => {
    const { asFragment } = ProductListComponent;
    expect(asFragment()).toMatchSnapshot();
  });
});
