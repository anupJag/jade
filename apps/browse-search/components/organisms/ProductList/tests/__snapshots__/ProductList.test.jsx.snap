// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList should render products correctly 1`] = `
<DocumentFragment>
  <div
    class="css-1vs4fqp"
  >
    <div
      class="css-1nujo00"
    >
      <div
        class="css-1tv17wi"
      >
        <a
          href="/p/184060007"
        >
          <h3
            class="css-tfj4pz"
          >
            Fresho Apple - Royal Gala, Regular
          </h3>
        </a>
        <div
          class="css-84pv12"
        >
          <img
            alt="Fresho Apple - Royal Gala, Regular"
            class="css-1phd9a0"
            height="150px"
            width="150px"
          />
        </div>
        <div
          class="css-gg4vpm"
        >
          <div
            class="css-fw0mfk"
          >
            $0.24
          </div>
          <div
            class="css-k008qs"
          >
            <svg
              class="css-10io2jg"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              class="css-10io2jg"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              class="css-1xb1rov"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              class="css-1xb1rov"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              class="css-1xb1rov"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <span
              class="css-1e92nls"
            >
              (100)
            </span>
          </div>
        </div>
        <div
          class="css-151y6vh"
        >
          A combination of slightly tart tasting skin and honey floral tasting flesh, the Royal Gala apples, as the name suggests looks regal with beautiful golden coloured streaks. 
        </div>
        <button
          class="css-bbsppe"
          type="button"
        >
          Add to Cart
        </button>
      </div>
    </div>
    <div
      class="css-1nujo00"
    >
      <div
        class="css-1tv17wi"
      >
        <a
          href="/p/960038585"
        >
          <h3
            class="css-tfj4pz"
          >
            Fresho Capsicum - Green, 500 g
          </h3>
        </a>
        <div
          class="css-84pv12"
        >
          <img
            alt="Fresho Capsicum - Green, 500 g"
            class="css-1phd9a0"
            height="150px"
            width="150px"
          />
        </div>
        <div
          class="css-gg4vpm"
        >
          <div
            class="css-fw0mfk"
          >
            $1.25
          </div>
          <div
            class="css-k008qs"
          >
            <svg
              class="css-10io2jg"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              class="css-10io2jg"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              class="css-10io2jg"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              class="css-10io2jg"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              class="css-10io2jg"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <span
              class="css-1e92nls"
            >
              (50)
            </span>
          </div>
        </div>
        <div
          class="css-151y6vh"
        >
          Leaving a moderately pungent taste on the tongue, Green capsicums, also known as green peppers are bell shaped, medium-sized fruit pods. They have thick and shiny skin with a fleshy texture inside. 
        </div>
        <button
          class="css-bbsppe"
          type="button"
        >
          Add to Cart
        </button>
      </div>
    </div>
    <div
      class="css-1nujo00"
    >
      <div
        class="css-1tv17wi"
      >
        <a
          href="/p/960106027"
        >
          <h3
            class="css-tfj4pz"
          >
            Fresho Kiwi - Green, 3 pcs
          </h3>
        </a>
        <div
          class="css-84pv12"
        >
          <img
            alt=""
            class="css-1phd9a0"
            height="150px"
            width="150px"
          />
        </div>
        <div
          class="css-gg4vpm"
        >
          <div
            class="css-fw0mfk"
          >
            $11.99
          </div>
          <div
            class="css-k008qs"
          >
            <svg
              class="css-10io2jg"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              class="css-10io2jg"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              class="css-10io2jg"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              class="css-10io2jg"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              class="css-1xb1rov"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <span
              class="css-1e92nls"
            >
              (100)
            </span>
          </div>
        </div>
        <div
          class="css-151y6vh"
        >
          Kiwis are oval shaped with a brownish outer skin. The flesh is bright green and juicy with tiny, edible black seeds. With its distinct sweet-sour taste and a pleasant smell, it tastes like strawberry and honeydew melon. 
        </div>
        <button
          class="css-bbsppe"
          type="button"
        >
          Add to Cart
        </button>
      </div>
    </div>
    <div
      class="css-1nujo00"
    >
      <div
        class="css-1tv17wi"
      >
        <a
          href="/p/184570092"
        >
          <h3
            class="css-tfj4pz"
          >
            Fresho Pomegranate, 4 pcs (approx. 800 to 880 g)
          </h3>
        </a>
        <div
          class="css-84pv12"
        >
          <img
            alt=""
            class="css-1phd9a0"
            height="150px"
            width="150px"
          />
        </div>
        <div
          class="css-gg4vpm"
        >
          <div
            class="css-fw0mfk"
          >
            $0.50
          </div>
          <div
            class="css-k008qs"
          >
            <svg
              class="css-10io2jg"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              class="css-10io2jg"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              class="css-10io2jg"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              class="css-10io2jg"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              class="css-10io2jg"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <span
              class="css-1e92nls"
            >
              (1)
            </span>
          </div>
        </div>
        <div
          class="css-151y6vh"
        >
          With ruby color and an intense floral, sweet-tart flavor, the pomegranate delivers both taste and beauty. You can remove the skin and the membranes to get at the delicious fruit with nutty seeds. Fresho Pomegranates are finely sorted and graded to deliver the best tasting pomegranates to you. 
        </div>
        <button
          class="css-bbsppe"
          type="button"
        >
          Add to Cart
        </button>
      </div>
    </div>
    <div
      class="css-1nujo00"
    >
      <div
        class="css-1tv17wi"
      >
        <a
          href="/p/184570092"
        >
          <h3
            class="css-tfj4pz"
          >
            Fresho Tomato - Local, Organically Grown, 500 g
          </h3>
        </a>
        <div
          class="css-84pv12"
        >
          <img
            alt=""
            class="css-1phd9a0"
            height="150px"
            width="150px"
          />
        </div>
        <div
          class="css-gg4vpm"
        >
          <div
            class="css-fw0mfk"
          >
            $4.99
          </div>
          <div
            class="css-k008qs"
          >
            <svg
              class="css-10io2jg"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              class="css-10io2jg"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              class="css-10io2jg"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              class="css-1xb1rov"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              class="css-1xb1rov"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <span
              class="css-1e92nls"
            >
              (100)
            </span>
          </div>
        </div>
        <div
          class="css-151y6vh"
        >
          Fresho brings to you an exquisite range of locally grown organic tomatoes, which are now available at bigbasket. These organic tomatoes are free from harmful pesticides and insecticides and thus have a very natural and tangy taste, which is attributed to its healthy soil growth. The ripe and firm tomatoes are a luscious addition to any meal.
        </div>
        <button
          class="css-bbsppe"
          type="button"
        >
          Add to Cart
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
